cms/lib/xblock/upstream_sync.py

```diff
@@ -311,7 +313,37 @@ def decline_sync(downstream: XBlock, user_id=None) -> None:
         store.update_item(downstream, user_id)
 
 
-def sever_upstream_link(downstream: XBlock) -> None:
+def _update_children_top_level_parent(downstream: XBlock, new_top_level_parent_key: XBlock) -> list[XBlock]:
```

Comment:
```suggestion
def _update_children_top_level_parent(downstream: XBlock, new_top_level_parent_key: dict[str, str] | None) -> list[XBlock]:
```

Or even better would be to create a new dict type that has `type` and `id` keys.

Reply:
We have the `BlockKey` type that represent it, so I updated the code to use it here: [cb40abb](https://github.com/openedx/edx-platform/pull/37215/commits/cb40abb4fa236dd2771adad9d8e841848f387df9)